askup/views.py

```diff
@@ -229,19 +234,29 @@ def get_queryset(self):
 @login_required
 def user_profile_view(request, user_id):
     """Provide the user profile view."""
-    user = get_object_or_404(User, pk=user_id)
+    profile_user = get_object_or_404(User, pk=user_id)
     return render(
         request,
         'askup/user_profile.html',
         {
-            'first_name': user.first_name,
-            'last_name': user.last_name,
-            'email': user.email,
-            'questions_count': get_user_questions_count(user.id),
-            'answers_count': get_user_answers_count(user.id),
-            'own_score': get_user_score_by_id(user.id),
-            'is_owner': user.id == request.user.id,
-            'is_student': check_user_has_groups(request.user, 'student')
+            'first_name': profile_user.first_name,
+            'last_name': profile_user.last_name,
+            'email': profile_user.email,
+            'own_profile': int(user_id) == request.user.id,
+            'own_score': get_user_score_by_id(profile_user.id),
+            'is_owner': profile_user.id == request.user.id,
+            'is_student': check_user_has_groups(request.user, 'student'),
```

Comment:
Is it still used?
And why you check group for user from request but not from profile?

Reply:
You right. 

- `245 line` - is a logical duplication for the `247 line` - I'll remove it and will use an old 247 line in a template
- `248 line` - will be used in the next two parts of the Student's dashboard (the "Rank list" and the "My questions")

Thank you.